src/towerside/sequence.hpp

```diff
@@ -0,0 +1,22 @@
+#ifndef SEQUENCE_H
+#define SEQUENCE_H
+
+#include "common/config.hpp"
+
+namespace sequence {
+
+enum State {
+    MANUAL,
+    AUTOMATIC,
+    SEQUENCE1,
+    SEQUENCE2,
+};
```

Comment:
You should decouple automatic/manual from which sequence is selected. Then whether we are in automatic or manual is just the direct switch value from clientside, and the only 'stateful' thing is a "NONE/SEQUENCE1/SEQUENCE2" for what is currently running.